### ISSUE TITLE
Floating panels (not only the script editor)

### ISSUE THREAD

**Issue by Frontrider** [godotengine/godot-proposals#8305]
### Describe the project you are working on

I'm working with the tileset editor right now, setting up auto tiling then checking with the tilemap if I did everything correctly.

### Describe the problem or limitation you are having in your project

There is not enough room for me to look through the tileset, but there would be more than enough room on my other screen.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Let me to pop out any of the (bottom) panels into a separate window the same way it is done for the script editor. This lets me utilize screen space more effectively without having to continuously resize panels inside the editor. 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Basically the same kind of functionality that we have in jetbrains editors.

![image](https://github.com/godotengine/godot-proposals/assets/6755034/1590b2a6-a09c-42e6-ab68-335bcbc03f5c)
Any panel can be undocked and moved to a separate window.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Can not be worked around.

### Is there a reason why this should be core and not an add-on in the asset library?

This is additional editor functionality.

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/28.
